cmd/goal/clerk.go

```diff
@@ -872,20 +875,12 @@ var compileCmd = &cobra.Command{
 				outblob = protocol.Encode(&ls)
 			}
 			if !noProgramOutput {
-				fout, err := os.Create(outname)
-				if err != nil {
-					reportErrorf("%s: %s\n", outname, err)
-				}
-				_, err = fout.Write(outblob)
-				if err != nil {
-					reportErrorf("%s: %s\n", outname, err)
-				}
-				err = fout.Close()
+				err := writeFile(outname, outblob, 0666)
 				if err != nil {
 					reportErrorf("%s: %s\n", outname, err)
 				}
 			}
-			if !signProgram {
+			if !signProgram && outname != stdoutFilenameValue {
```

Comment:
If the filename is stdoutFilenameValue and the use specified signProgram, we need to exit with an error code.

Reply:
I think this is okay. I think this is just saying "if we are signing the program, we don't need to print its hash because it's not going to be used as a contract account because we're signing it for delegate useage"; and it's saying that we don't have a dedicated output "--hash-out-file foo" so if stdout is in use we don't print the program hash.

Reply:
If we were asked by the user to provide the signature of the program, than we should either do that, or completely fail. What about changing the output format so that it will be considered a TEAL comment ?

Reply:
I'm wondering if two things are getting mixed up on this thread.
1. the hash of the program, the sha512_256 of the program used as a 32 byte Address of a contract-account. printed on line 886 `fmt.Printf("%s: %s\n", fname, addr.String())`
2. a LogicSig where the program has been ed25519 signed by the private key of a standard account to make a delegated LogicSig, assembled in the `if signProgram{}` block 854-876.
The hash isn't needed if we're in signProgram/LogicSig mode; and is dropped if stdout is being used for the regular output.

Reply:
Ahh I see.
Yes, I missed the two `if signProgram` and `if !signProgram` sections. I thought that you've just ignoring the user request completely which seems incorrect.